machineconfiguration/v1/types_machineosbuild.go

```diff
@@ -0,0 +1,208 @@
+package v1
+
```

Comment:
Based on the discussions in the HCP office hours, this will work in the same way that MachineConfigs would through configmap-mapped objects.

The main reasoning being that we would like to decouple management cluster version and the API, such that e.g. a 4.17 management cluster can configure it for a 4.19 hosted cluster, despite the API not being available for 4.17

Reply:
Have you considered an alternative abstraction at the HCP layer that would be namespace scoped? It would look very similar to this API, but expose it in a way that makes sense for HCP.

Is there any intention that, in HCP, these builds exist within the management plane?

Reply:
>The main reasoning being that we would like to decouple management cluster version and the API, such that e.g. a 4.17 management cluster can configure it for a 4.19 hosted cluster, despite the API not being available for 4.17

I think setting a req on the management cluster version for certain features is valid. Besides the API how will the mco controllers watch and react to those configmaps? What Joel suggest above is one possible starting point to answer these questions. Let's make sure we iterate a bit more on this before setting an API in stone.

Reply:
>Besides the API how will the mco controllers watch and react to those configmaps?